### PR TITLE
Resolve bug in API View Script

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Resolve Bug in Create APIView Script

### CODE REVIEW

File: eng/scripts/Create-APIView.ps1 [hunk 1]
```diff
@@ -111,15 +117,17 @@ function Invoke-SwaggerAPIViewParser {
       LogGroupEnd
 
       $generatedAPIViewTokenFile = Get-ChildItem -File | Select-Object -First 1
```
Comment (weshaggard) [inline eng/scripts/Create-APIView.ps1:119]: Do we not know the expected outputted file name?

File: eng/scripts/Create-APIView.ps1 [hunk 2]
```diff
@@ -111,15 +117,17 @@ function Invoke-SwaggerAPIViewParser {
...
-      $readMeTag = $generatedAPIViewTokenFile.BaseName
-
-      LogSuccess " Generated '$Type' APIView Token File using file, '$readMeFile' and tag '$readMeTag'"
 
-      $apiViewTokensFilePath = [System.IO.Path]::Combine($TokenDirectory, "$resourceProvider.$Type.json")
-      LogInfo " Moving generated APIView Token file to '$apiViewTokensFilePath'"
-      Move-Item -Path $generatedAPIViewTokenFile.FullName -Destination $apiViewTokensFilePath -Force > $null
+      if (Test-Path -Path $generatedAPIViewTokenFile) {
```
Comment (weshaggard) [inline eng/scripts/Create-APIView.ps1:121]: If this file doesn't exist I would assume the parser failed. Wouldn't we have a failing exit code to look at in that case?

File: eng/scripts/Create-APIView.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/scripts/Create-APIView.ps1:119]: Is the file named after the tag? If so is that a hacky way of getting the tag name as output or does it make since for it to be named that and we should get that value another way?

File: eng/scripts/Create-APIView.ps1 [hunk 3]
```diff
@@ -183,12 +191,12 @@ function New-SwaggerAPIViewTokens {
...
+      $swaggerReadMeFiles.Add($readmeFile) | Out-Null
     }
   }
 
   LogGroupStart " Swagger APIView Tokens will be generated for the following configuration files..."
-  $readmeFile | ForEach-Object {
+  $swaggerReadMeFiles | ForEach-Object {
     LogInfo " - $_"
   }
   LogGroupEnd
```
Comment (weshaggard) [inline eng/scripts/Create-APIView.ps1:202]: Below are we assuming the that default tag name doesn't change between the source and the target? What if they change the tag in the PR?

File: eng/scripts/Create-APIView.ps1 (hunk 1 above)
Comment (chidozieononiwu) [reply eng/scripts/Create-APIView.ps1:119]: Yes the output file is name is known. The output file names if the name of the tag used to create the token file.

File: eng/scripts/Create-APIView.ps1 (hunk 2 above)
Comment (chidozieononiwu) [reply eng/scripts/Create-APIView.ps1:121]: No necessarily. In the case where the tag is not found in the readme.md it will not create any APIView token.

File: eng/scripts/Create-APIView.ps1 (hunk 3 above)
Comment (chidozieononiwu) [reply eng/scripts/Create-APIView.ps1:201]: I have updated the logic to revert to using the default tag if the new tag (source branch) is not found on the target branch

File: eng/scripts/Create-APIView.ps1 [hunk 4]
```diff
@@ -322,4 +339,45 @@ function New-RestSpecsAPIViewReviews {
...
+  The file path to the readme file.
+#>
+function Get-TagInformationFromReadMeFile {
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]$ReadMeFilePath
+  )
+  $tags = [System.Collections.Generic.HashSet[string]]::new()
+  $markDownContent = Get-Content -Path $ReadMeFilePath
+  $chheckForDefaulttag = $false
```
Comment (weshaggard) [inline eng/scripts/Create-APIView.ps1:358]:
```suggestion
  $checkForDefaulttag = $false
```

File: eng/scripts/Create-APIView.ps1 [hunk 5]
```diff
@@ -322,4 +339,45 @@ function New-RestSpecsAPIViewReviews {
...
+
+  foreach ($line in $markDownContent) {
+    if ($line -match "### Basic Information") {
+      $chheckForDefaulttag = $true
+    }
+
+    if ($chheckForDefaulttag -and ($null -eq $defaultTag)) {
+      if ($line -match "^tag:\s(.+)") {
+        $defaultTag = $matches[1]
+        $chheckForDefaulttag = $false
```
Comment (weshaggard) [inline eng/scripts/Create-APIView.ps1:369]:
```suggestion
        $checkForDefaulttag = $false
```

File: eng/scripts/Create-APIView.ps1 [hunk 6]
```diff
@@ -322,4 +339,45 @@ function New-RestSpecsAPIViewReviews {
...
+  $defaultTag = $null
+
+  foreach ($line in $markDownContent) {
+    if ($line -match "### Basic Information") {
+      $chheckForDefaulttag = $true
+    }
+
+    if ($chheckForDefaulttag -and ($null -eq $defaultTag)) {
+      if ($line -match "^tag:\s(.+)") {
+        $defaultTag = $matches[1]
```
Comment (weshaggard) [inline eng/scripts/Create-APIView.ps1:368]:
```suggestion
        $defaultTag = $matches["tag"].Value
```

File: eng/scripts/Create-APIView.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/scripts/Create-APIView.ps1:121]: Don't we know the name of the file now? Why are we still calling Get-ChildItem here?

File: eng/scripts/Create-APIView.ps1 [hunk 7]
```diff
@@ -163,7 +170,7 @@ function New-SwaggerAPIViewTokens {
 
   # Get Changed Swagger Files
   LogInfo " Getting changed swagger files in PR, between $SourceCommitId and $TargetCommitId"
-  $changedFiles = Get-ChangedFiles -baseCommitish $SourceCommitId -targetCommitish $TargetCommitId
+  $changedFiles = Get-ChangedFiles
```
Comment (weshaggard) [inline eng/scripts/Create-APIView.ps1:173]: Why did you change this now? Are you now only looking at the list of files changes in the PR?

Comment (chidozieononiwu) [reply eng/scripts/Create-APIView.ps1:173]: Right. I am allowing this to use the default parameters which is using `HEAD^ & HEAD` which is the merge commit and the target branch commit. That seems to be the right way to get files changed in the PR.

However the swagger tool itself is run on the source branch commit and then the target branch

File: eng/scripts/Create-APIView.ps1 [hunk 8]
```diff
@@ -322,4 +339,45 @@ function New-RestSpecsAPIViewReviews {
...
+    if ($checkForDefaulttag -and ($null -eq $defaultTag)) {
+      if ($line -match "^tag:\s*(?<tag>.+)") {
+        $defaultTag = $matches["tag"]
+        $checkForDefaultTag = $false
+      }
+    }
+
+    if ($line -match '^```\s*yaml\s*\$\(tag\)\s*==\s*''(?<tag>.+)''') {
+      $tag = $matches["tag"]
+      $tags.Add($tag) | Out-Null
```
Comment (weshaggard) [inline eng/scripts/Create-APIView.ps1:375]: out of curiosity why do you need `Out-Null` here?

**Review by weshaggard — APPROVED** [review]

File: eng/scripts/Create-APIView.ps1 (hunk 8 above)
Comment (chidozieononiwu) [reply eng/scripts/Create-APIView.ps1:375]: It outputs True or False depending on if the command succeeded.
